### DISCUSSION TITLE
Warning in console when use in Nuxt 3 project.but everything worked fine.

### DISCUSSION THREAD

**Discussion by Monirhasanmunna** [vueform/vueform#227] [Questions]
here is the warning:
ssr:warn [Vue warn]: Failed to resolve component: Vueform
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
    at getStack (C)

how to solve this??
![Screenshot_3](https://github.com/vueform/vueform/assets/71868847/c319c528-bd88-45b7-870c-35b2a8cbb93e)

**Comment by davidparys** [comment]
Same here, seems like they're not auto imported and so you can't actually use them as in VSCode auto complete. No type completions either.

**Comment by ghyath5** [comment]
Doesn't it support ssr ?

**Comment by efernau** [comment]
Same here, 
later with `generate` I run into ssr hydration problems.

 `ClientOnly` Disable SSR altogether for the problem area. 🙃

```vue
<template>
    <ClientOnly>
      <Vueform v-bind="form" @error="handleError" @success="handleSuccess" />
    </ClientOnly>
</template>
```
I don't think that's a good thing, but it does remove the nasty warnings.

**Comment by hosseinjarrahi** [comment]
i have this error yet, i disable the ssr but still has error

**Comment by spencerjsmall** [comment]
Bumping this

**Comment by unreadableusername** [comment]
I have the same problem with `nuxt:3.14.159` where vueform triggers a page long warning in the console every time a page containing a form is rendered on the serverside. Sadly I cannot just wrap it in ClientOnly tags because I need to update the form values and the ref is not accessible in a onMounted() hook when wrapped like suggested above.

Short excerpt from the warning in the console:
```
 WARN  [Vue warn]: Component <Anonymous> is missing template or render function. 
  at <Anonymous ref="form$" endpoint="/api/submit-application" method="post" > 
  at <[vacancy] onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <Anonymous key="/application-form/abc()" vnode= { __v_isVNode: true,
  __v_skip: true,

(...)

 WARN  [nuxt] Failed to stringify dev server logs. Received DevalueError: Cannot stringify a function. You can define your own reducer/reviver for rich types following the instructions in https://nuxt.com/docs/api/composables/use-nuxt-app#payload.
```

**Comment by felixxxxxs** [comment]
Still no solution to this issue?

**Comment by JamieCurnow** [comment]
I've "fixed" this in nuxt by adding a little SSR-only plugin:

```ts
// app/plugins/vueform-ssr-stub.ts
// fix the SSR warning for Vueform on SSR
export default defineNuxtPlugin((nuxtApp) => {
  if (import.meta.server) {
    const stub = { render: () => null }
    nuxtApp.vueApp.component('Vueform', stub)
    nuxtApp.vueApp.component('VueformElement', stub)
  }
})
```

The issue was an SSR issue because the VueForm nuxt module does:
https://raw.githubusercontent.com/vueform/vueform/main/nuxt/src/module.ts:

```ts
if (import.meta.client) {
  // ...
  nuxtApp.vueApp.use(vueform, vueformConfig)
}
```

Which means no component get registerred in SSR. This is required because VueForm relies on client side stuff.

PS. with this fix you still need to remember to wrap `<VueForm>` in `<ClientOnly>` 👍
